sdk/core/core-http/lib/nodeFetchHttpClient.ts

```diff
@@ -22,8 +23,57 @@ if (typeof globalWithFetch.fetch !== "function") {
 }
 
 export class NodeFetchHttpClient extends FetchHttpClient {
+  private httpAgent: http.Agent | undefined;
+  private httpsAgent: https.Agent | undefined;
+  private httpClientOptions: HttpClientOptions;
+
   private readonly cookieJar = new tough.CookieJar(undefined, { looseMode: true });
 
+  constructor(httpClientOptions?: HttpClientOptions) {
+    super();
+
+    this.httpClientOptions = {
+      keepAlive: true,
+      proxySettings: undefined,
+      ...httpClientOptions
+    };
+  }
+
+  private getOrCreateAgent(httpRequest: WebResource): http.Agent | https.Agent {
+    let agent: http.Agent | https.Agent | undefined = undefined;
+    const isHttps = isUrlHttps(httpRequest.url);
+
+    if (isHttps) {
+      agent = this.httpsAgent;
+    } else {
+      agent = this.httpAgent;
+    }
+
+    if (!agent) {
+      if (this.httpClientOptions.proxySettings) {
+        const tunnel: ProxyAgent = createProxyAgent(
+          httpRequest.url,
+          this.httpClientOptions.proxySettings,
+          httpRequest.headers
+        );
+
+        agent = tunnel.agent;
+        if (tunnel.isHttps) {
+          this.httpsAgent = tunnel.agent as https.Agent;
+        } else {
+          this.httpAgent = tunnel.agent;
+        }
+      } else {
+        const agentOptions: http.AgentOptions | https.AgentOptions = {
+          keepAlive: this.httpClientOptions.keepAlive
+        };
+        agent = isHttps ? new https.Agent(agentOptions) : new http.Agent(agentOptions);
```

Comment:
Do you want to store `agent` on `this` here as well? Otherwise it looks like you'll be creating a new agent each time `prepareRequest` is called.

Reply:
Damn it, thanks for catching that!